docker-style/action.yml

```diff
@@ -0,0 +1,160 @@
+      2: info)
+    required: false
+    default: 2
+    type: int
+
+runs:
+  using: "composite"
+  steps:
+
+    # ------------------------------------------------------------------------
+
+    - uses: ansys/actions/_logging@main
+      with:
+        level: "INFO"
+        message: >
+          Check that directory 'docker' exists at the root of the repository.
+          If it does not, then a warning is emitted.
+
+    - name: "Check if docker directory exists"
+      shell: bash
+      run: |
+        if [ -d "${{ github.workspace }}/docker" ]; then
+          echo "HAS_DOCKER_DIR=true" >> $GITHUB_ENV
+        else
+          echo "HAS_DOCKER_DIR=false" >> $GITHUB_ENV
+        fi
+
+    # ------------------------------------------------------------------------
+
+    - uses: ansys/actions/_logging@main
+      if: env.HAS_DOCKER_DIR == 'false'
+      with:
+        level: "WARNING"
+        message: >
+          No 'docker' directory found. If possible, please follow the common
+          approach of storing docker files in the 'docker' directory at the
+          root of the repository.
+
+    # ------------------------------------------------------------------------
+
+    - uses: ansys/actions/_logging@main
+      with:
+        level: "INFO"
+        message: >
+          Check that directory input exist. If not, a warning is raised and
+          non existing ones (if any) are ignored.
+
+    - name: "Filter input directory"
+      shell: bash
+      run: |
+        EXISTING_DIRS=""
+        NON_EXISTING_DIRS=""
+        for directory in ${{ inputs.directory }}
+        do
+          if [ -d "${{ github.workspace }}/$directory" ]; then
+            EXISTING_DIRS+="$directory "
+          else
+            NON_EXISTING_DIRS+="$directory "
+          fi
+        done
+        EXISTING_DIRS=$(echo -n "$EXISTING_DIRS" | xargs)
+        echo "EXISTING_DIRS=$EXISTING_DIRS" >> $GITHUB_ENV
+        NON_EXISTING_DIRS=$(echo -n "$NON_EXISTING_DIRS" | xargs)
+        echo "NON_EXISTING_DIRS=$NON_EXISTING_DIRS" >> $GITHUB_ENV
+
+    # ------------------------------------------------------------------------
+
+    - uses: ansys/actions/_logging@main
+      if: ${{ !!env.EXISTING_DIRS }}
+      with:
+        level: "INFO"
+        message: >
+          Lint '${{ inputs.directory }}'.
+
+    - name: "Format Hadolint dockerfile input"
+      if: ${{ !!env.EXISTING_DIRS }}
+      shell: bash
+      run: |
+        fmt_directory="${{ inputs.directory }}"
+        if [[ "${{ inputs.recursive }}" == 'true' ]]; then
+          fmt_directory_array=($fmt_directory)
+          for i in "${!fmt_directory_array[@]}"; do
+            fmt_directory_array[i]="${fmt_directory_array[i]}/**/Dockerfile"
+          done
+          fmt_directory="${fmt_directory_array[*]}"
+        else
+          fmt_directory="${directories}/Dockerfile"
+        fi
+        echo "FORMATED_HADOLINT_DOCKERFILE_INPUT=$fmt_directory" >> $GITHUB_ENV
```

Comment:
What happens if ``inputs.recursive`` is false??  then you only analyse ``"${directories}/Dockerfile"``... where ``directories`` is defined??

Reply:
Oh daman you are right, nice catch here !

Reply:
@germa89 now everything is working fine. Btw, I also found another bug while fixing your remark :/